File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        res=[-1]*len(arr)
        print(res)
        right=len(arr)-2
        mx=arr[-1]
        while right>=0:
            res[right]=mx
            mx=max(mx,arr[right])
            right-=1
        return res

        
       
                


            
                
        

File: 1231-replace-elements-with-greatest-element-on-right-side/1231-replace-elements-with-greatest-element-on-right-side.ts
function replaceElements(arr: number[]): number[] {
    let array_length=arr.length;

    let maxNumber:number=arr[array_length-1]
    arr[array_length-1]=-1
    for(let index=arr.length-2;index>=0;index--){
        let temp:number=arr[index]
        arr[index]=maxNumber
        maxNumber=Math.max(temp,maxNumber)
    }
    return arr

    
};